libkmod/libkmod-index.c

```diff
@@ -270,7 +269,12 @@ static struct index_node_f *index_read(FILE *in, uint32_t offset)
 		strbuf_init(&buf);
 		while (value_count--) {
 			priority = read_long(in);
-			buf_freadchars(&buf, in);
+			if (buf_freadchars(&buf, in)) {
```

Comment:
preferred style is to check < 0 for errors when an int is returned (which is also needed when you add back the > 0).

Reply:
Adjusted